### DISCUSSION TITLE
Getting ResourceInUseException when creating tables.

### DISCUSSION THREAD

**Discussion by Karan-TR** [spulec/moto#7606] [Q&A]
Using boto3==1.28.62, botocore==1.31.62, urllib3==1.26.15 and moto[all]==5.0.5

@pytest.fixture
def dynamodb_session():
    with mock_aws():
        dynamo_db = boto3.resource("dynamodb", region_name="us-east-1")
        yield dynamo_db

I am using this session to create tables in multiple tests with same name. 
I keep getting error saying Table already exists. It works if i explicitly delete the tables, but according to documentation Moto should handle deleting the resources after every test.

**Comment by bblommers** [comment]
Hi @Karan-TR, is that fixture session-scoped? If it is, `pytest` will only start the mock once per session, so then you would indeed run into the fact that data is persisted across tests.

You can use `@pytest.fixture(scope="function")` to make it explicit that this fixture should be re-executed for every test-function.

**Reply by Karan-TR** [reply]
Hey!

So, I read about it and default scope is always function and I tried explicitly mentioning scope as function but didnt help.

**Reply by bblommers** [reply]
Can you share a stand-alone, completely reproducible test case?

It should indeed work, but it's hard to tell what's going wrong based on just this fixture.
